types/mojang-minecraft-server-admin/index.d.ts

```diff
@@ -0,0 +1,66 @@
+// Type definitions for Minecraft Bedrock Edition script APIs (experimental) 0.1
+// Project: https://docs.microsoft.com/minecraft/creator/
+// Definitions by: Jake Shirley <https://github.com/JakeShirley>
+//                 Mike Ammerlaan <https://github.com/mammerla>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/* *****************************************************************************
+   Copyright (c) Microsoft Corporation.
+   ***************************************************************************** */
+/**
+ * This represents a placeholder object that represents a
+ * secret string. The contents of that string are not available
+ * to script; this object is just a placeholder.
+ */
+export class SecretString {
+    constructor(value: string);
+    equals(other: SecretString): boolean;
+}
+/**
+ * A collection of server secrets defined in dedicated server
+ * configuration.
+ */
+export class ServerSecrets {
+    /**
+     * A list of available, configured server secrets.
+     */
+    readonly 'names': string[];
+    /**
+     * @remarks
+     * Returns a SecretString that is a placeholder for a secret
+     * configured in a JSON file. In certain objects, like an
+     * HttpHeader, this Secret is resolved at the time of execution
+     * but is not made available to the script environment.
+     * @param name
+     */
+    get(name: string): SecretString;
+    protected constructor();
```

Comment:
what's the use of `protected` constructor here (and other place)? It will prevent usage, is that intended?

Reply:
Perhaps there is a better way to do this but I'll describe what the idea is:

These types describe our interface of natively created (by the runtime in our game) types and modules.  Most of these types do not get a constructor bound, so calls to create them are not valid and would fail.  This was my attempt to hint to anyone using TS that you should not be trying to `new` these types anywhere in script-land.

If there is a better way to do this I'd love to know :) I am not a TS expert by any means.

Reply:
No, that's ok for me. For luxon `private` is used: #59971
Interfaces can be used to just describe non-constructable contracts

Reply:
👍🏽 I can look at using `private`.  Interfaces are in our backlog but we don't current support duck-typing for our native JS bindings (which is a bummer!).